optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/LambdaBeanPropertyMemberAccessor.java

```diff
@@ -76,7 +76,10 @@ private Function createGetterFunction(MethodHandles.Lookup lookup) {
                     MethodType.methodType(Object.class, Object.class),
                     lookup.findVirtual(declaringClass, getterMethod.getName(), MethodType.methodType(propertyType)),
                     MethodType.methodType(propertyType, declaringClass));
-        } catch (LambdaConversionException | NoSuchMethodException | IllegalAccessException e) {
+        } catch (IllegalAccessException e) {
+            throw new IllegalStateException("Lambda creation failed for getterMethod (" + getterMethod + "). You may " +
+                    " consider supplying a ClassLoader parameter to your SolverFactory.create...() method call.", e);
```

Comment:
Good change, this will help a lot of spring users. One thing for consistency:
For fail fast exception messages, optaplanner-core puts suggestions or conjectures on a second line (so after a newline) starting with the word "Maybe" (even if it's 90% sure) to signal that that could solve the user's problem but we're not a 100% sure.
Also nice to have is active voice (remove "consider", remove "your"), see https://developers.google.com/style/voice

In this case that would give:

            throw new IllegalStateException("Lambda creation failed for getterMethod (" + getterMethod + ").\n" +
                    "Maybe add getClass().getClassLoader() as a parameter to the " + SolverFactory.class.getSimpleName() + ".create...() method call.", e);

Thinking aloud: Should we mention "spring boot" in this exception message explicitly? Probably not - if a user runs into this - interrupting their flow - the maybe will set them back on track in a minute - (much better than the hours they loose now).